src/bin/cargo/cli.rs

```diff
@@ -180,14 +210,27 @@ fn print_list(gctx: &GlobalContext, is_verbose: bool) {
             CommandInfo::External { path } => {
                 if let Some(desc) = known_external_desc {
                     drop_println!(gctx, "    {literal}{name:<20}{literal:#} {desc}");
-                } else if is_verbose {
-                    drop_println!(
-                        gctx,
-                        "    {literal}{name:<20}{literal:#} {}",
-                        path.display()
-                    );
                 } else {
-                    drop_println!(gctx, "    {literal}{name}{literal:#}");
+                    drop_print!(gctx, "    {literal}{name:<20}{literal:#}");
+
+                    // The .unwrap() in the path below will probably never panic, since we are dealing with files here
+                    let installed_by = user_installed
+                        .get(&path.file_name().unwrap().to_string_lossy().to_string())
+                        .cloned();
+
+                    if let Some(installed_by) = installed_by {
+                        let desc = registry.description(&installed_by).ok().flatten();
```

Comment:
Personally, I feel like it's overkill talking over the network for descriptions. Yet, let's back to design discussion first.

Reply:
Querying the network for such little things is overkill indeed. However, from what I've found,currently no other better way to do this exists.
Perhaps what could be done here is implement some form of cache to store binaries' descriptions so that we don't have to query them each time.